tests/test_activate.py

```diff
@@ -1227,18 +1228,22 @@ def test_set_default_start_env(self):
         activator.deactivate()
         exit_code_3 = main_subshell("remove", "-n", "test-env", "--all", "-y")
         exit_code_4 = main_subshell(
-            "config", "--remove-key", "default_start_environment"
+            "config", "--remove-key", "auto_activate_environment"
         )
         context.__init__()
 
-    def test_default_start_env_base_if_not_set(self):
+    def test_auto_activate_env_base_if_not_set(self):
         # activate an unspecified env
-        activator = PosixActivator(["activate"])
+        activator = (PowerShellActivator(["activate", "--no-stack"]) if on_win else PosixActivator(["activate", "--no-stack"]))
         activator.execute()
 
         # if the we've managed to activate the base environment, the test passes
         assert activator.env_name_or_prefix == "base"
 
+        # isolate the test by clearing out the env and resetting the context
+        activator.deactivate()
+        context.__init__()
+
```

Comment:
I think we have some helpers to ensure the context is reset properly in `conda.testing.*`. That said, I don't think `activate()` or `deactivate()` really have effects on a running Python process? The shell code needs to be eval'd in the shell, usually. What's the concern here?